### DISCUSSION TITLE
What is the expression range (positive cutoff) of imputed protein and RNA levels in multimodal reference mapping scCITE-seq & RNA-seq?

### DISCUSSION THREAD

**Discussion by denvercal1234GitHub** [satijalab/seurat#5058] [Q&A]
Hi there, 

Related to the Issue #4240 , I mapped my RNA data onto CITE-seq ref of PBMC and imputed the protein level (https://satijalab.org/seurat/articles/multimodal_reference_mapping.html). 

```
## Find avg expression of RNA
DefaultAssay(Seurat_obj) <- 'RNA'

AverageExpression(Seurat_obj, assays = 'RNA', features = "CD4")

## Find avg expression of imputed protein
DefaultAssay(Seurat_obj) <- 'predicted_ADT'

AverageExpression(Seurat_obj, assays = 'predicted_ADT', features = "CD4-1")

```

Would anyone mind helping me with 5 questions below? Thank you so much for your help!

**1. What is the range of possible values for the imputed protein level? Does this range depend on my own data or intrinsic to the CITE-seq ref data? Same questions for sc-RNA expression level.** 

I want to know the range because I aim to find which cells have "reasonable" protein expression of CD4 and CD8 in cluster 20 for instance? In other words, what would be a positive cut-off of protein expression and RNA expression for a marker in CITE data and in scRNA-seq data? Using WhichCell() setting `CD4-1` > 0 would simply give me all the cells in my data. 

```
DefaultAssay(Seurat_obj) <- 'predicted_ADT'

WhichCells(Seurat_obj, expression = `CD4-1` > 0 & `CD8` <= 0 & `CD8a` <= 0 & seurat_clusters == "20")
```

Expression levels across clusters shown as example:

<img width="599" alt="Screen Shot 2021-09-08 at 12 29 12 PM" src="https://user-images.githubusercontent.com/62179082/132501594-3d185097-a407-412a-b477-e96d514b2d9e.png">

<img width="565" alt="Screen Shot 2021-09-08 at 12 30 14 PM" src="https://user-images.githubusercontent.com/62179082/132501719-c4d3eb3e-6e2a-4e21-a8c2-bba2344ff1d2.png">

**2. Is there a way to get the imputed protein expression data for every cell for a given gene in a cluster as a data frame?**

**3. What is the difference between CD8 and CD8a in the CITE-seq ref data?**

**4. Once I set DefaultAssay() to 'RNA' then I switched back setting DefaultAssay() to 'predicted_ADT', why the active assay now would remain 'RNA'? When I compute AverageExpression(), does setting the argument assay='predicted_ADT' is sufficient without setting DefaultAssay()?** 

<img width="573" alt="Screen Shot 2021-09-08 at 12 35 51 PM" src="https://user-images.githubusercontent.com/62179082/132502450-cd58131f-6c15-41d7-bd02-3d6c90eff548.png">

**5. There is a drastic difference between the RNA expression and protein expression for this particular marker CD4. Has anyone else observed similar discrepancy for this marker between RNA and protein? What are some other markers known to have such difference?**

**Comment by timoast** [comment]
> 1. What is the range of possible values for the imputed protein level? Does this range depend on my own data or intrinsic to the CITE-seq ref data? Same questions for sc-RNA expression level.

The possible range is the same as the range from the original CITE-seq data that's being imputed, since it is a weighted average.

> 2. Is there a way to get the imputed protein expression data for every cell for a given gene in a cluster as a data frame?

You can use the `GetAssayData()` function to extract the expression matrix, or the `AverageExpression()` function to get the average expression per group of cells (eg, cluster)

> 3. What is the difference between CD8 and CD8a in the CITE-seq ref data?

CD8 is the protein, CD8A is the gene

> 4. Once I set DefaultAssay() to 'RNA' then I switched back setting DefaultAssay() to 'predicted_ADT', why the active assay now would remain 'RNA'? When I compute AverageExpression(), does setting the argument assay='predicted_ADT' is sufficient without setting DefaultAssay()?

Your assignment arrow is pointing in the wrong direction

> 5. There is a drastic difference between the RNA expression and protein expression for this particular marker CD4. Has anyone else observed similar discrepancy for this marker between RNA and protein? What are some other markers known to have such difference?

Yes, this is very commonly observed in CITE-seq data. The protein signal is much more robust than the RNA, and is one of the reasons why CITE-seq is so useful in studying immune cell populations.

**Reply by denvercal1234GitHub** [reply]
Thank you @timoast for your response. 

So, for CITE-seq data, at which value should I call a cell being positive for a marker? For example, above 0? above 3SD of the mean across all cells?)? 

For scRNA-seq data, I just call a marker having expression count > 0 after normalization & scaling as positive for a marker (ie WhichCells(seurat_object, expression = `CD4` > 0). Is it right?

Thank you again!
